pkg/tfbridge/provider.go

```diff
@@ -517,7 +517,7 @@ func buildTerraformConfig(ctx context.Context, p *Provider, vars resource.Proper
 		}
 	}
 
-	inputs, _, err := MakeTerraformInputs(ctx, nil, tfVars, nil, tfVars, p.config, p.info.Config)
+	inputs, _, err := makeTerraformInputsWithoutTFDefaults(ctx, nil, tfVars, nil, tfVars, p.config, p.info.Config)
```

Comment:
Note I've now changed `CheckConfig` to not get the TF defaults - this addresses half of https://github.com/pulumi/pulumi-terraform-bridge/issues/1603 and gets the right behaviour WRT empty MaxItemsOne config

Reply:
I've had to revert this - it doesn't work. Instead I added a `makeTerraformInputsWithoutMaxItemsOneDefaults` variant which is used here - it applies TF defaults but not the maxItemsOne empty arrays.